ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -386,6 +386,24 @@ class PartCentredForceHolder {
     }
   }
 
+  // If there is a target vessel, and that vessel is known to the plugin,
+  // returns it.  Otherwise returns null.
+  private Vessel TargetVessel() {
+    Vessel target = FlightGlobals.fetch.VesselTarget?.GetVessel();
+    string target_id = target?.id.ToString();
+    if (target_id != null && plugin_.HasVessel(target_id)) {
+      return target;
+    } else {
+      return null;
+    }
+  }
+
+  // Same as above but returns the GUID.
+  private string TargetVesselGuid() {
+    return TargetVessel().id.ToString();
```

Comment:
This throws NullReferenceException if `TargetVessel()` is null; did you mean `TargetVessel()?.id.ToString()`?